File: examples/python/client/client_clear.py
"""
Irondomo Protocol client example. Uses the IDPClient API to hide all IDP aspects
Author: Matteo Ferrabone <matteo.ferrabone@gmail.com> 
"""
import os
import sys
from IronDomo  import IDPClient
import time
import begin
import uuid

@begin.start
def main(service='echo', identity='echo'):
    verbose = '-v' in sys.argv
    client = IDPClient.IronDomoClient("tcp://localhost:6555", verbose, identity=identity+uuid.uuid4().hex)
    count = 0
    loop = True
    request = bytearray(os.urandom(1000000))
    while count < 10000:
        #request = "Hello world {0} -> {1}".format(service, count)
        try:
            reply = client.send(service.encode(), request)#[request.encode(), b'parte2'])
            print('Message: {0}'.format(count))
        except KeyboardInterrupt:
            break
        else:
            # also break on failure to reply:
            if reply is None:
                break
        count += 1
    print("%i requests/replies processed" % count)

File: C/examples/client_clear.c
//
//  Irondomo Protocol  CLEAR client example
//  Uses the idcli API to hide all IDP aspects
//

//  Lets us build this source without creating a library
#include "idcliapi.h"

int main (int argc, char *argv [])
{
    int verbose = (argc > 1 && streq (argv [1], "-v"));
    idcli_t *session = idcli_new ("tcp://localhost:5000", "EchoClient", verbose);

    idcli_connect_to_broker(session);

    int count;
    for (count = 0; count < 2000; count++) {
        zmsg_t *request = zmsg_new ();
        zmsg_pushstr (request, "Hello");
        zmsg_pushstr (request, " ");
        zmsg_pushstr (request, "world");
        zmsg_t *reply = idcli_send (session, "echo", &request);
        if (reply)
            zmsg_destroy (&reply);
        else
            break;              //  Interrupt or failure
    }
    printf ("%d requests/replies processed\n", count);
    idcli_destroy (&session);
    return 0;
}
